### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by fdreyer** [JetsGame/gLund#3]
Step 1:
- [x] Create some preprocessing pipeline for the lund images: PCA for dimensionality reduction (autoencoder), etc. Define some wrapper around sklearn.preprocessing.
- [x] Create framework to swap out between all the different generative models, with proper handling of image (for convnet) or projected data (for dense net)
- [x] Write a main program that reads set of Lund images, uses some preprocessing if requested, and trains one of the generative models.
- [x] Get some sensible 2d plots using gan/wgan/wgangp/vae

Step 2:
- [x] Build framework to read in and write out the models, etc.
- [x] Find good hyperparameters with hyperopt. Loss function: L_2 norm or KL of difference between between average Lund image of 50k pythia jets and 50k generated jets
- [x] For the paper: train on detector level samples => we need to create some Delphes pythia samples for QCD jets.

Final diagnostic plots we want:
- [x] Average lund image for 50k jets for all generators, as well as ratio plots to pythia
- [x] 1-d plot of number of activated pixels per image for each setup, comparing to pythia
- [x] 1-d distribution of a slice in kt and of a slice in delta R for each setup, comparing to pythia
